crates/prover-telemetry/src/lib.rs

```diff
@@ -0,0 +1,205 @@
+    // Those metrics are not linked to any registry
+    pub static ref SEND_TX: opentelemetry::metrics::Counter<u64> = global::meter(AGGLAYER_RPC_OTEL_SCOPE_NAME)
+        .u64_counter("send_tx")
+        .with_description("Number of transactions received on the RPC")
+        .build();
+
+    pub static ref VERIFY_ZKP: opentelemetry::metrics::Counter<u64> = global::meter(AGGLAYER_KERNEL_OTEL_SCOPE_NAME)
+        .u64_counter("verify_zkp")
+        .with_description("Number of ZKP verifications")
+        .build();
+
+    pub static ref VERIFY_SIGNATURE: opentelemetry::metrics::Counter<u64> = global::meter(AGGLAYER_KERNEL_OTEL_SCOPE_NAME)
+        .u64_counter("verify_signature")
+        .with_description("Number of signature verifications")
+        .build();
+
+    pub static ref CHECK_TX: opentelemetry::metrics::Counter<u64> = global::meter(AGGLAYER_KERNEL_OTEL_SCOPE_NAME)
+        .u64_counter("check_tx")
+        .with_description("Number of transactions checked")
+        .build();
+
+    pub static ref EXECUTE: opentelemetry::metrics::Counter<u64> = global::meter(AGGLAYER_KERNEL_OTEL_SCOPE_NAME)
+        .u64_counter("execute")
+        .with_description("Number of transactions executed")
+        .build();
+
+    pub static ref SETTLE: opentelemetry::metrics::Counter<u64> = global::meter(AGGLAYER_KERNEL_OTEL_SCOPE_NAME)
+        .u64_counter("settle")
+        .with_description("Number of transactions settled")
+        .build();
+}
+
+pub mod prover {
+    use lazy_static::lazy_static;
+    use opentelemetry::global;
+
+    use crate::constant::AGGLAYER_PROVER_RPC_OTEL_SCOPE_NAME;
+
+    lazy_static! {
+        pub static ref PROVING_REQUEST_RECV: opentelemetry::metrics::Counter<u64> =
+            global::meter(AGGLAYER_PROVER_RPC_OTEL_SCOPE_NAME)
+                .u64_counter("proving_request_recv")
+                .with_description("Number of proving request received")
+                .build();
+        pub static ref PROVING_REQUEST_SUCCEEDED: opentelemetry::metrics::Counter<u64> =
+            global::meter(AGGLAYER_PROVER_RPC_OTEL_SCOPE_NAME)
+                .u64_counter("proving_request_succeeded")
+                .with_description("Number of proving request that succeeded")
+                .build();
+        pub static ref PROVING_REQUEST_FAILED: opentelemetry::metrics::Counter<u64> =
+            global::meter(AGGLAYER_PROVER_RPC_OTEL_SCOPE_NAME)
+                .u64_counter("proving_request_failed")
+                .with_description("Number of proving request that failed")
+                .build();
+        pub static ref PROVING_FALLBACK_TRIGGERED: opentelemetry::metrics::Counter<u64> =
+            global::meter(AGGLAYER_PROVER_RPC_OTEL_SCOPE_NAME)
+                .u64_counter("proving_fallback_triggered")
+                .with_description("Number of proving fallback triggered")
+                .build();
+    }
+}
+
+pub struct ServerBuilder {}
```

Comment:
Nit: Could this be an empty type?

It does not appear to be actually created anywhere but maybe `buildsctructor` does so internally? (Though I don't know how would it do that.)

```suggestion
pub enum ServerBuilder {}
```